### PR TITLE
fix: skip this.errors.remove if useUnderlyingErrorsValue

### PR DESCRIPTION
**Pull request by spham92**
Skip the code flow for `_removeError ` if `this._schema.useUnderlyingErrorsValue(this._modelName)` returns true

### CODE REVIEW

File: tests/unit/model/errors-attribute-test.js [hunk 1]
```diff
@@ -89,6 +95,25 @@ module('unit/model/errors-attribute', function (hooks) {
...
+          id: 'urn:book:1',
+          type: 'com.example.bookstore.Book',
+          attributes: {
+            author: ['urn:author:1'],
+          },
+        },
+      });
+    });
+    const getErrorsSpy = this.sinon.spy(model, 'errors', ['get']);
+    model._removeError('author');
```
Comment (igorT) [inline tests/unit/model/errors-attribute-test.js:113]: These kinds of tests couple us to current specific implementation. If we refactor `_removeError` to a different method, this test will fail, even though the behavior might still be correct. Can we write this test to assert user observable behavior?

**Review by igorT — CHANGES_REQUESTED** [review]

File: tests/unit/model/errors-attribute-test.js (hunk 1 above)
Comment (spham92) [reply tests/unit/model/errors-attribute-test.js]: Ended changing to use `model.setUnknownProperty('title', 'Harry Potter');` which calls `_removeError`

Comment (igorT) [reply tests/unit/model/errors-attribute-test.js]: It seems like you are still testing the internal implementation though. The user visible behavior here, is that if `useUnderlyingErrorsValue` is `true` then you can access `errors` after setting a property. If it's false, and you set a prop, the error gets removed, right?

Comment (igorT) [reply tests/unit/model/errors-attribute-test.js]: Lets also create the errors using the public api and not by setting the property directly.  The `save errors getting updated via the store and removed upon setting a new value` has an example of how to get invalid errors. I've cleaned up that test as the setup wasn't quite correct here: https://github.com/hjdivad/ember-m3/pull/1251

**Review by igorT — APPROVED** [review]

**Pull request merged**